internal/core/application/operator/fee_fragmenter_account.go

```diff
@@ -0,0 +1,198 @@
+package operator
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/tdex-network/tdex-daemon/internal/core/domain"
+	"github.com/tdex-network/tdex-daemon/internal/core/ports"
+	"github.com/vulpemventures/go-elements/address"
+)
+
+const (
+	maxNumFragments     = 20
+	defaultNumFragments = maxNumFragments
+	feeFragmentAmount   = uint64(5000)
+)
+
+func (s *service) DeriveFeeFragmenterAddresses(
+	ctx context.Context, num int,
+) ([]string, error) {
+	if !s.accountExists(ctx, domain.FeeFragmenterAccount) {
+		if _, err := s.wallet.Account().CreateAccount(
+			ctx, domain.FeeFragmenterAccount,
+		); err != nil {
+			return nil, err
+		}
+	}
+	return s.wallet.Account().DeriveAddresses(
+		ctx, domain.FeeFragmenterAccount, num,
+	)
+}
+
+func (s *service) ListFeeFragmenterExternalAddresses(
+	ctx context.Context,
+) ([]string, error) {
+	return s.wallet.Account().ListAddresses(ctx, domain.FeeFragmenterAccount)
+}
+
+func (s *service) GetFeeFragmenterBalance(
+	ctx context.Context,
+) (map[string]ports.Balance, error) {
+	return s.wallet.Account().GetBalance(ctx, domain.FeeFragmenterAccount)
+}
+
+func (s *service) FeeFragmenterSplitFunds(
+	ctx context.Context, maxFragments uint32, millisatsPerByte uint64,
+	chRes chan ports.FragmenterReply,
+) {
+	defer close(chRes)
+
+	chRes <- fragmenterReply{"fetching fee fragmenter account LBTC balance", nil}
+
+	balanceInfo, err := s.wallet.Account().GetBalance(ctx, domain.FeeFragmenterAccount)
+	if err != nil {
+		chRes <- fragmenterReply{"", err}
+		return
+	}
+	if len(balanceInfo) <= 0 {
+		chRes <- fragmenterReply{
+			"", fmt.Errorf("fee fragmenter account has 0 LBTC balance"),
+		}
+		return
+	}
+	lbtc := s.wallet.NativeAsset()
+	lbtcBalance, ok := balanceInfo[lbtc]
+	if !ok || lbtcBalance.GetConfirmedBalance() == 0 {
+		chRes <- fragmenterReply{
+			"", fmt.Errorf("fee fragmenter account has 0 LBTC balance"),
+		}
+		return
+	}
+
+	balance := lbtcBalance.GetConfirmedBalance()
+	if balance < feeFragmentAmount {
+		chRes <- fragmenterReply{
+			"", fmt.Errorf(
+				"not enough LBTC balance  (%d) to fragment (required ar least %d)",
+				balance, feeFragmentAmount,
+			),
+		}
+		return
+	}
+	numFragments := defaultNumFragments
+	if maxFragments > 0 {
+		numFragments = int(maxFragments)
+	}
+	if feeFragmentAmount*uint64(numFragments)-1 >= balance {
+		chRes <- fragmenterReply{
+			"", fmt.Errorf(
+				"not enough lbtc balance (%d) for requested number of fragments "+
+					"(%d - %d sats each)",
+				lbtcBalance.GetConfirmedBalance(), numFragments, feeFragmentAmount,
+			)}
+		return
+	}
+
+	chRes <- fragmenterReply{
+		fmt.Sprintf(
+			"fetched funds with total LBTC balance of %d, to be split into %d "+
+				"fragments of %d sats each", balance, numFragments, feeFragmentAmount,
+		), nil,
+	}
+
+	// make sure the fee account is created.
+	s.wallet.Account().CreateAccount(ctx, domain.FeeAccount)
```

Comment:
I see in many places errors were not checked, this looks problematic even if we are sure error wont happen or it is not important, for eg. here we are calling other process(ocean) and in case ocean is not reachable error will be "silent" and we wont know.
To find all places where error is not handled use golangci-lint run --disable-all -E errcheck

Reply:
done.